daemon/rename.go

```
@@ -21,6 +21,10 @@ func (daemon *Daemon) ContainerRename(oldName, newName string) error {
 		return fmt.Errorf("Neither old nor new names may be empty")
 	}
 
+        if oldName == newName {
```

We should probably error out instead of returning nil.